extension/src/controllers/subtitle-controller.ts

```diff
@@ -113,36 +118,46 @@ export default class SubtitleController {
             return;
         }
 
-        this._resetOverlays(this._subtitleAlignment, preCacheDom);
+        this._resetOverlays(preCacheDom);
         this._preCacheDom = preCacheDom;
     }
 
     cacheHtml() {
-        if (!(this.subtitlesElementOverlay instanceof CachingElementOverlay)) {
+        if (!(this.bottomSubtitlesElementOverlay instanceof CachingElementOverlay) || !(this.topSubtitlesElementOverlay instanceof CachingElementOverlay)) {
```

Comment:
Currently, the check assumes that both overlays are always the same type, which may be true and enforced internally by this class.

But since it does not hurt, I think to be safe the `instanceof` check should be done independently for `bottomSubtitlesElementOverlay` and `topSubtitlesElementOverlay` before calling `uncacheHtml`.